takwimu/views.py

```diff
@@ -206,18 +215,30 @@ def _extract_search_results(self, request, page):
         (country, category) = (str(page.get_parent()), page.title) \
             if isinstance(page, ProfileSectionPage) \
             else (str(page), u'Country Overview')
-        self.countries[country] = 1
-        self.topics[category] = 1
-        url = page.get_url(request)
-        for topic in page.body:
-            result = {
-                'country': country,
-                'region': 'National',
-                'category': category,
-                'url': url,
-                'data_point': topic,
-            }
-            self.items.append(result)
+        if self._filter_results(country, self.country_filter) and self._filter_results(category, self.topic_filter):
```

Comment:
Use filters in search itself, otherwise we're doing what elasticsearch should be doing.